packages/electron-updater/src/AppUpdater.ts

```diff
@@ -482,6 +483,22 @@ export abstract class AppUpdater extends (EventEmitter as new () => TypedEmitter
     }
   }
 
+  // delete the pending files to save space
+  private async deletePendingUpdate() {
+    if (this.downloadedUpdateHelper?.cacheDirForPendingUpdate) {
+      try {
+        const files = await fs.promises.readdir(this.downloadedUpdateHelper.cacheDirForPendingUpdate)
+        for (const file of files) {
+          await fs.promises.unlink(path.join(this.downloadedUpdateHelper.cacheDirForPendingUpdate, file))
```

Comment:
Few notes:
- Why `unlink` instead of `fsExtra.remove(...)`?
- Why not just delete the entire dir instead of each file within it? (First should also check if dir exists in case an error )
- We can't use `fs.promises` in `electron-updater` in order to support legacy (and very deprecated) versions of electron w/o releasing a breaking change to electron-updater. To keep this as a `patch` fix, please use `fs-extra`, otherwise this is a `major` semver bump

Reply:
Fixed. Please help review again.